Moonshot/Combat/Stats.gd

```diff
@@ -69,10 +69,11 @@ func pickup_firerate(increase):
 		
 func take_damage(level_of_decrease):
 	if modifiers['firerate_pickups'] > 0:
-		modifiers['firerate_pickups'] -= no_firerate_pickups_to_increase_firerate*level_of_decrease
+		modifiers['firerate_pickups'] -= no_firerate_pickups_to_increase_firerate
 		current_firerate_level = floor(modifiers['firerate_pickups'] / no_firerate_pickups_to_increase_firerate) + 1
 		modifiers['firerate'] = pow(firerate_scaling_factor, current_firerate_level)
 		emit_signal("firerate_changed", modifiers['firerate_pickups'], current_firerate_level)
+	health -= level_of_decrease
```

Comment:
I don't think there should be 2 take_damage functions, split across player and stats

Reply:
We should also get rid of invulnerable in this file actully, its not used anywhere.

Reply:
...so have this func called decrease_firerate? And move `health -= level_of_decrease` to its own function in stats, or back to player.take_damage?

Reply:
Ithink there should definitly be a wrapper inside stats for taking damage as there may be other effects that it needs to handle. At the same time the player needs to have a take damage function of some description to handle and animation. What i'll do is create a seperate function inside stats for decreasing firerate and call it from the stats.take_damage function so that its clearer what is happening. I mean, we should really have a signal emitted by the player when the player takes damage to decouple them- but don't want to do this at the moment. Will come back to this PR tomorrow.